backend/local/metadata.go

```diff
@@ -129,10 +129,13 @@ func (o *Object) writeMetadataToFile(m fs.Metadata) (outErr error) {
 	}
 	mode, hasMode := o.parseMetadataInt(m, "mode", 8)
 	if hasMode {
-		if mode >= 0 && uint(mode) <= math.MaxUint32 {
-			err = os.Chmod(o.path, os.FileMode(mode))
-			if err != nil {
-				outErr = fmt.Errorf("failed to change permissions: %w", err)
+		if mode >= 0 {
+			umode := uint(mode)
+			if umode <= math.MaxUint32 {
+				err = os.Chmod(o.path, os.FileMode(umode))
```

Comment:
It's complaining about the o.path not the FileMode here. That data is supplied by the user, but the Paths in the show Paths are nuts!

So I think this new alert is a false positive

Reply:
Yeah, it is not even new, the same o.path was reported previously as well:
- https://github.com/rclone/rclone/security/code-scanning/136

And there are currently 33 reported "Uncontrolled data used in path expression"...